### PR TITLE
Fixed the issue with rotating near borders or another blocks

### PR DESCRIPTION
**Pull request by Arrinao**
I just hope we won't end up on really_new_borderline_rotate branch

### CODE REVIEW

File: tetris.py [hunk 1]
```diff
@@ -222,8 +222,19 @@ def block_rotator(self):
         Rotates the current block
         """
         self.rotate_counter += 1
+        if any(x <= -1 for (x, y) in self.get_current_block()) or any(
+            (x - 1, y) in self.landed_blocks for x, y in self.get_current_block()
+        ):
+            self.rotate_counter -= 1
+        if any(x >= game_width for x, y in self.get_current_block()) or any(
+            (x + 1, y) in self.landed_blocks for x, y in self.get_current_block()
+        ):
```
Comment (Akuli) [inline tetris.py:231]:
```suggestion
        if any(x <= -1 or x >= game_width for (x, y) in self.get_current_block()) or any(
            (x, y) in self.landed_blocks for x, y in self.get_current_block()
        ):
```

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py [hunk 2]
```diff
@@ -222,8 +222,15 @@ def block_rotator(self):
         Rotates the current block
         """
         self.rotate_counter += 1
+        if any(x <= -1 or x >= game_width for (x, y) in self.get_current_block()) or any(
+            (x, y) in self.landed_blocks for x, y in self.get_current_block()
+        ):
```
Comment (Akuli) [inline tetris.py:227]:
```suggestion
        if any(x not in range(game_width) or (x, y) in self.landed_blocks for (x, y) in self.get_current_block()):
```

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py (hunk 2 above)
Comment (Arrinao) [reply tetris.py:227]: Would you mind me keeping the above code commented out?

The last one is nice but it's something I wouldn't come up with at the moment :)

Comment (Akuli) [reply tetris.py:227]: We can keep it as is. It already works.

**Review by Akuli — APPROVED** [review]
Now `black` and we can merge.

**Pull request merged**